### PR TITLE
gh-76183: Fix legacy locale coercion tests on platforms that already have a default C.UTF-8 locale

### PR DESCRIPTION
**Pull request by embray**
Fix legacy locale coercion tests on platforms that already have a default C.UTF-8 locale

* BPO issue: bpo-32002

* Issue: gh-76183

### CODE REVIEW

File: Lib/test/test_c_locale_coercion.py [hunk 1]
```diff
@@ -181,6 +183,22 @@ def setUpModule():
         CLI_COERCION_TARGET = AVAILABLE_TARGETS[0]
         CLI_COERCION_WARNING = CLI_COERCION_WARNING_FMT.format(CLI_COERCION_TARGET)
 
+    # Determine the platform's default LC_CTYPE from the default locale.
+    # POSIX provides that there *exists* a default locale to provide when
+    # no enivornment variables are set, or are set to empty strings.  However,
```
Comment (ncoghlan) [inline Lib/test/test_c_locale_coercion.py]: Typo in the comment: enivornment -> environment

Comment (embray) [reply Lib/test/test_c_locale_coercion.py]: Heh, I'll just fix that and squash...

File: Lib/test/test_c_locale_coercion.py [hunk 2]
```diff
@@ -181,6 +183,22 @@ def setUpModule():
...
+    result, py_cmd = run_python_until_end(
+        "-c", "import locale; print(locale.setlocale(locale.LC_CTYPE))",
+        __isolated=True,
+        LANG="", LC_ALL="", LC_CTYPE="", PYTHONCOERCECLOCALE="0"
+    )
+    if not result.rc == 0:
+        result.fail(py_cmd)
+
+    DEFAULT_LC_CTYPE = result.out.decode("ascii").strip().upper()
+
```
Comment (xdegaye) [inline Lib/test/test_c_locale_coercion.py:201]: Did you try setting `C_LOCALE_STREAM_ENCODING`  to `"utf-8"` instead ?

File: Lib/test/test_c_locale_coercion.py [hunk 3]
```diff
@@ -301,6 +319,11 @@ def _check_c_locale_coercion(self,
                 # See https://bugs.python.org/issue30672 for discussion
                 if locale_to_set == "POSIX":
                     continue
+                elif locale_to_set == "" and DEFAULT_LC_CTYPE == "C.UTF-8":
```
Comment (xdegaye) [inline Lib/test/test_c_locale_coercion.py:322]: The condition is too restrictive:
 * When `locale_to_set` is `""` and  a locale envt variable exists in `extra_vars` with a non empty string (for example in `test_LC_ALL_set_to_C`), the subtest may be run.
* It seems that when  `C_LOCALE_STREAM_ENCODING` is set to `"utf-8"` and `coerce_c_localecoerce_c_locale` is not `"warn"` then the test should pass.

**Review by xdegaye — CHANGES_REQUESTED** [review]

**Comment by erlend-aasland** [comment]
@embray: are you planning on following up this PR/issue?

**Comment by erlend-aasland** [comment]
After one month, still no response. Suggesting to close this PR and the related issue (Cygwin is not supported).

**Comment by arhadthedev** [comment]
@embray ping?

**Pull request closed without merging**